src/QMCWaveFunctions/tests/test_hybridrep.cpp

```diff
@@ -238,6 +238,13 @@ TEST_CASE("Hybridrep SPO from HDF diamond_2x1x1", "[wavefunction]")
   ions_.update();
   elec_.update();
 
+  ParticleSet::RealType r;
+  ParticleSet::PosType dr;
+  elec_.getDistTable(0).get_first_neighbor(0, r, dr, false);
+  std::cout << std::setprecision(14) << "check r^2 against dr^2. "
+            << "r= " << r << " dr = " << dr << std::endl;
+  REQUIRE(r * r == dot(dr, dr));
```

Comment:
Realistically I think a couple of machine epsilon of tolerance is needed here.

Reply:
Even before the change `std::numeric_limits<ParticleSet::RealType>::epsilon()` passes.
Are you comfortable with `std::numeric_limits<double>::epsilon()`?

Reply:
Yes. Not a big deal. I am surprised we haven't run into this problem with other compilers/optimized math implementations, since taking shortcuts in sqrt is a common optimization.